x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,128 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+	}
+	if cf.Expiry.Before(sdkCtx.HeaderInfo().Time) {
+		return sdk.Coin{}, fmt.Errorf("cannot withdraw continuous funds\ncontinuous fund expired for recipient: %s", recipient.String())
+	}
+
+	toDistributeAmount, err := k.ToDistribute.Get(ctx)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	if !toDistributeAmount.Equal(math.ZeroInt()) {
+		err = k.iterateAndUpdateFundsDistribution(ctx, toDistributeAmount)
+		if err != nil {
+			return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+		}
+	}
+
+	// withdraw continuous fund
+	withdrawnAmount, err := k.withdrawRecipientFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while withdrawing recipient funds for recipient: %s", recipient.String())
+	}
+
+	return withdrawnAmount, nil
+}
+
+func (k Keeper) withdrawRecipientFunds(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	// get allocated continuous fund
+	fundsAllocated, err := k.getAndResetDistributedFunds(ctx, recipient)
```

Comment:
reset should happen after a successful send, reseting and failing to may cause some issues